### PR TITLE
feat: add advanced options for random poem

### LINKED ISSUE #144: feat: advanced random poem

### PR DESCRIPTION
**Pull request by MuhammadKhosravi**
resolves #144

### CODE REVIEW

File: app/src/main/java/ir/jaamebaade/jaamebaade_client/view/components/AdvancedRandomPoemOptions.kt [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                poetsWithCategories?.let { categories ->
+                    RecursiveList(categories = categories)
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun RecursiveList(
```
Comment (ErfanMojibi) [inline app/src/main/java/ir/jaamebaade/jaamebaade_client/view/components/AdvancedRandomPoemOptions.kt:60]: `RecursiveCheckList` could be a better name?

File: app/src/main/java/ir/jaamebaade/jaamebaade_client/view/components/AdvancedRandomPoemOptions.kt
Comment (ErfanMojibi) [inline app/src/main/java/ir/jaamebaade/jaamebaade_client/view/components/AdvancedRandomPoemOptions.kt:1]: Removeandvanced from name? :)

File: app/src/main/java/ir/jaamebaade/jaamebaade_client/view/components/AdvancedRandomPoemOptions.kt [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import ir.jaamebaade.jaamebaade_client.R
+import ir.jaamebaade.jaamebaade_client.viewmodel.DownloadedPoetViewModel
+import ir.jaamebaade.jaamebaade_client.wrapper.CategoryGraphNode
+
+@Composable
+@OptIn(ExperimentalMaterial3Api::class)
+fun AdvancedRandomPoemOptions(
```
Comment (ErfanMojibi) [inline app/src/main/java/ir/jaamebaade/jaamebaade_client/view/components/AdvancedRandomPoemOptions.kt:26]: remove `advanced` from name?

File: app/src/main/java/ir/jaamebaade/jaamebaade_client/view/components/AdvancedRandomPoemOptions.kt [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        }
+    ) {
+        Column(
+            modifier = modifier
+                .padding(10.dp)
+                .verticalScroll(rememberScrollState())
+        ) {
+            Text(
+                text = stringResource(id = R.string.ADVANCED_RANDOM_POEM_OPTIONS),
+                style = MaterialTheme.typography.titleMedium
```
Comment (ErfanMojibi) [inline app/src/main/java/ir/jaamebaade/jaamebaade_client/view/components/AdvancedRandomPoemOptions.kt:47]: again remove advanced from name

File: app/src/main/java/ir/jaamebaade/jaamebaade_client/view/components/Topbar.kt [hunk 4]
```diff
@@ -66,15 +68,21 @@ fun IntroShowcaseScope.TopBar(
                 && backStackEntry?.destination?.route != AppRoutes.FAVORITE_SCREEN.toString())
 
     val context = LocalContext.current
+    val sheetState = rememberModalBottomSheetState()
+    var showAdvancedRandomPoemOptions by remember { mutableStateOf(false) }
```
Comment (ErfanMojibi) [inline app/src/main/java/ir/jaamebaade/jaamebaade_client/view/components/Topbar.kt:72]: remove advanced from name?

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -56,6 +56,8 @@
     <string name="DELETE">حذف</string>
     <string name="SEARCH">جست‌وجو</string>
     <string name="CLOSE">بستن</string>
+    <string name="RANDOM_POEM">شعر رندم</string>
+    <string name="ADVANCED_RANDOM_POEM_OPTIONS">تنظیمات پیشرفته شعر تصادفی</string>
```
Comment (ErfanMojibi) [inline app/src/main/res/values/strings.xml:60]: تنظیمات شعر تصادفی

File: app/src/main/java/ir/jaamebaade/jaamebaade_client/view/components/AdvancedRandomPoemOptions.kt (hunk 1 above)
Comment (MuhammadKhosravi) [reply app/src/main/java/ir/jaamebaade/jaamebaade_client/view/components/AdvancedRandomPoemOptions.kt:60]: done

File: app/src/main/java/ir/jaamebaade/jaamebaade_client/view/components/AdvancedRandomPoemOptions.kt
Comment (MuhammadKhosravi) [reply app/src/main/java/ir/jaamebaade/jaamebaade_client/view/components/AdvancedRandomPoemOptions.kt:1]: done

File: app/src/main/java/ir/jaamebaade/jaamebaade_client/view/components/AdvancedRandomPoemOptions.kt (hunk 2 above)
Comment (MuhammadKhosravi) [reply app/src/main/java/ir/jaamebaade/jaamebaade_client/view/components/AdvancedRandomPoemOptions.kt:26]: done

File: app/src/main/java/ir/jaamebaade/jaamebaade_client/view/components/AdvancedRandomPoemOptions.kt (hunk 3 above)
Comment (MuhammadKhosravi) [reply app/src/main/java/ir/jaamebaade/jaamebaade_client/view/components/AdvancedRandomPoemOptions.kt:47]: done

File: app/src/main/java/ir/jaamebaade/jaamebaade_client/view/components/Topbar.kt (hunk 4 above)
Comment (MuhammadKhosravi) [reply app/src/main/java/ir/jaamebaade/jaamebaade_client/view/components/Topbar.kt:72]: done

File: app/src/main/res/values/strings.xml (hunk 5 above)
Comment (MuhammadKhosravi) [reply app/src/main/res/values/strings.xml:60]: done

**Pull request merged**